index.html

```diff
@@ -630,7 +630,43 @@ <h2>DID Documents</h2>
     </section>
 
     <section>
-      <h2>DID Resolvers and DID Resolution</h2>
+      <h2>DID Resolvers</h2>
+
+      <p>
+A <a>DID resolver</a> is a software and/or hardware component that takes a
+<a>DID</a> as input and produces a <a>conforming DID document</a> as output.
+This process is called <a>DID resolution</a> and often includes resolution input
+options and resolution output metadata.
+      </p>
+      <p>
+The general process of <a>DID resolution</a> is described in
+<a href="#did-resolution"></a>. The details of implementing a
+<a>DID resolver</a> are described in the DID Resolution specification
+[[?DID-RESOLUTION]].
+      </p>
+    </section>
+
+    <section>
+      <h2>DID Dereferencers</h2>
+
+      <p>
+A <a>DID URL dereferencer</a> is a software and/or hardware component that takes
+a <a>DID URL</a> as input and produces a resource, which might or might not be a
+<a>conforming DID Document</a> as output. This process is called <a>DID URL
+dereferencing</a> and often includes dereferencing input options and
+dereferencing output metadata. Generally, a <a>DID URL dereferencer</a> uses a
```

Comment:
Same comment as above: Suggest include input and output metadata in the definition. Phrasing "often includes" implies much more variability than is intended here.

```suggestion
a <a>DID URL</a> as input (and associated options) and produces a resource (and
associated metadata) as output in a process called <a>DID URL
dereferencing</a>. The resource produced by <a>DID URL dereferencing</a> might or might not be a
<a>conforming DID Document</a>. Generally, a <a>DID URL dereferencer</a> uses a
```